src/CSS Gradients/main.vala

```diff
@@ -0,0 +1,117 @@
+    update_css_provider (css);
+}
+
+void update_css_provider (string css) {
+    var display = Gdk.Display.get_default ();
+    if (css_provider == null) {
+        css_provider = new Gtk.CssProvider ();
+    } else {
+        Gtk.StyleContext.remove_provider_for_display (display, css_provider);
+    }
+
+    css_provider.load_from_string (css);
+    Gtk.StyleContext.add_provider_for_display (display, css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+}
+
+string generate_css () {
+    double angle = spin_row_angle.get_value ();
+    string angle_string = angle.to_string ();
+    string first_color_string = button_color_1.get_rgba ().to_string ();
+    string second_color_string = button_color_2.get_rgba ().to_string ();
+    string third_color_string = button_color_3.get_rgba ().to_string ();
+
+    uint selected = combo_row_gradient_type.selected;
+    if (selected == 0) {
+        css = """.background-gradient {
+    background-image: linear-gradient(
+        %sdeg,
+        %s,
+        %s,
+        %s
+    );
+}""".printf (angle_string, first_color_string, second_color_string, third_color_string);
+    } else if (selected == 1) {
+        css = """.background-gradient {
+    background-image: radial-gradient(
+        %s,
+        %s,
+        %s
+    );
+}""".printf (first_color_string, second_color_string, third_color_string);
+    } else if (selected == 2) {
+        css = """.background-gradient {
+    background-image: conic-gradient(
+        from %sdeg,
+        %s,
+        %s,
+        %s
+    );
+}""".printf (angle_string, first_color_string, second_color_string, third_color_string);
+    }
+    return css;
+}
+
+void update_color_scheme () {
+    string scheme_name = style_manager.dark ? "Adwaita-dark" : "Adwaita";
+    var scheme = scheme_manager.get_scheme (scheme_name);
+
+    if (scheme != null) {
+        gtksource_buffer.style_scheme = scheme;
+    }
+}
+
+public void main () {
+    combo_row_gradient_type = (Adw.ComboRow) workbench.builder.get_object ("combo_row_gradient_type");
+    spin_row_angle = (Adw.SpinRow) workbench.builder.get_object ("spin_row_angle");
+    button_color_1 = (Gtk.ColorDialogButton) workbench.builder.get_object ("button_color_1");
+    button_color_2 = (Gtk.ColorDialogButton) workbench.builder.get_object ("button_color_2");
+    button_color_3 = (Gtk.ColorDialogButton) workbench.builder.get_object ("button_color_3");
+    gtksource_buffer = (GtkSource.Buffer) workbench.builder.get_object ("gtksource_buffer");
+    var button_copy_css = (Gtk.Button) workbench.builder.get_object ("button_copy_css");
+
+    combo_row_gradient_type.notify["selected"].connect (update);
+    spin_row_angle.notify["value"].connect (update);
+    button_color_1.notify["rgba"].connect (update);
+    button_color_2.notify["rgba"].connect (update);
+    button_color_3.notify["rgba"].connect (update);
+/*
+ *  code view
+ */
+    var clipboard = Gdk.Display.get_default ().get_clipboard ();
```

Comment:
```suggestion
    Gdk.Clipboard clipboard = Gdk.Display.get_default ().get_clipboard ();
```
also here not var, if you cannot deviate the type from the assignment, it should be explicit